### PR TITLE
Added scale and interpolation to particles.

### PR DESCRIPTION
**Pull request by HalalUr**

### CODE REVIEW

File: src/render/particle_engine.cpp [hunk 1]
```diff
@@ -10,6 +10,7 @@
 #include <boost/lambda/bind.hpp>
 #include <boost/lambda/lambda.hpp>
 #include <boost/ref.hpp>
+#include "boost/algorithm/clamp.hpp"
```
Comment (MrPepperoni) [inline src/render/particle_engine.cpp:13]: <>

File: src/render/particle_system.cpp [hunk 2]
```diff
@@ -24,8 +25,14 @@ void ParticleSystem::Update( double DeltaTime )
...
         Actor& actor = **it;
         Opt<IEmitterComponent> emitterC = actor.Get<IEmitterComponent>();
-        Opt<IPositionComponent> pos = actor.Get<IPositionComponent>();
-        if ( !emitterC.IsValid() || !pos.IsValid() )
+        Opt<IPositionComponent> positionC = actor.Get<IPositionComponent>();
+        Opt<IMoveComponent> moveC = actor.Get<IMoveComponent>();
+        glm::vec2 speed(0);
+        if (moveC.IsValid()&&moveC->IsMoving()&&!moveC->IsRooted())
+        {
+            speed = glm::vec2(moveC->GetSpeedX()*DeltaTime, moveC->GetSpeedY()*DeltaTime);
```
Comment (MrPepperoni) [inline src/render/particle_system.cpp:33]: that's not speed, that's distance

File: src/render/particle_engine.cpp (hunk 1 above)
Comment (HalalUr) [inline src/render/particle_engine.cpp:13]: True sry thx.

File: src/render/particle_system.cpp (hunk 2 above)
Comment (HalalUr) [inline src/render/particle_system.cpp:33]: Will rename it.

**Pull request merged**